stage1/enterexec/enterexec.c

```diff
@@ -177,34 +177,30 @@ void initialize_keep_env(const char *keep_env_file, const char **keep_env)
 /* Try to set current env from keep_env and env file. */
 static void set_env(const char *env_file)
 {
-	struct stat st;
-	char *map, *k, *v;
-	typeof(st.st_size) i;
-
-	map_file(env_file, PROT_READ|PROT_WRITE, MAP_PRIVATE, &st, (void **)&map);
-
-	if(!st.st_size)
-		return;
-
-	map[st.st_size - 1] = '\0'; /* ensure the mapping is null-terminated */
-
-	for(i = 0; i < st.st_size;) {
-		k = &map[i];
-		i += strlen(k) + 1;
-		exit_if((v = strchr(k, '=')) == NULL,
-			"Malformed environment entry: \"%s\"", k);
-		/* a private writable map is used permitting s/=/\0/ */
+	FILE 		*f;
+	char 		*line = NULL;
+	size_t 		len = 0;
+	ssize_t 	read;
+	char 		*v;
+
+    pexit_if((f = fopen(env_file, "r")) == NULL,
```

Comment:
Formatting's out of whack - can you use spaces to stay consistent with the rest of the file?